apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/server/partners-table.tsx

```diff
@@ -0,0 +1,110 @@
+import { ConfirmationDialog } from "@/components/ui/ConfirmationDialog";
+import { Button } from "@/components/ui/button";
+import { cn } from "@/lib/utils";
+import { Trash2 } from "lucide-react";
+import Link from "next/link";
+import type { Ecosystem, Partner } from "../../../types";
+import { useDeletePartner } from "../../hooks/use-delete-partner";
+import { usePartners } from "../../hooks/use-partners";
+
+export function PartnersTable({ ecosystem }: { ecosystem: Ecosystem }) {
+  const { partners } = usePartners({ ecosystemId: ecosystem.id });
+
+  return (
+    <table className="w-full">
+      <thead>
+        <tr className="border-b bg-muted">
+          <TableHeading>Name</TableHeading>
+          <TableHeading>Domains</TableHeading>
+          <TableHeading>Bundle ID</TableHeading>
+          <TableHeading>Partner Key</TableHeading>
+          <TableHeading>Permissions</TableHeading>
+          {/* Empty space for delete button */}
+          <th />
+        </tr>
+      </thead>
+      <tbody>
+        {partners?.map((partner: Partner) => (
+          <TableRow key={partner.id} partner={partner} ecosystem={ecosystem} />
+        ))}
+      </tbody>
+    </table>
+  );
+}
+
+function TableHeading(props: {
+  children: React.ReactNode;
+  className?: string;
+}) {
+  return (
+    <th
+      className={cn(
+        "text-xs p-4 font-semibold tracking-wide text-left uppercase text-muted-foreground",
+        props.className,
+      )}
+    >
+      {props.children}
+    </th>
+  );
+}
+
+function TableRow(props: {
+  partner: Partner;
+  ecosystem: Ecosystem;
+}) {
+  const { deletePartner, isLoading: isDeleting } = useDeletePartner();
+
+  return (
+    <tr
+      className={cn(
+        "relative border-b hover:bg-secondary",
+        isDeleting && "animate-pulse",
+      )}
+    >
+      <TableData>{props.partner.name}</TableData>
+      <TableData>{props.partner.allowlistedDomains.join(", ")}</TableData>
+      <TableData>{props.partner.allowlistedBundleIds.join(", ")}</TableData>
+      <TableData>{props.partner.id}</TableData>
+      <TableData>{props.partner.permissions}</TableData>
+      <td className="py-1">
+        <ConfirmationDialog
+          title={`Are you sure you want to delete the partner ${props.partner.name}?`}
+          description={
+            <span>
+              Their partner key will no longer be able to use your ecosystem
+              wallet. Their users will still have access to their assets at{" "}
+              <Link
+                href={`https://${props.ecosystem.slug.split(".")[1]}.ecosystem.thirdweb.com`}
```

Comment:
why is this split necessary?

Reply:
The ecosystem "slug" (how partners specify the ecosystem in the SDK) is `ecosystem.name` and the domain for the wallet page is `name.ecosystem.thirdweb.com`, the split it to pull off the leading `ecosystem` from the slug (slug is not a good name for that identifier but its what the backend calls it)

Reply:
okay, that explains it, that said it feels _weird_ that "ecosystem" is baked into the "slug" instead of inferred, like I would expect the "slug" defined (in database) to be the "id" so say the ecosystem is "xai" I would expect the slug/id (id makes more sense to me as a name for this) to be "xai" not "ecosystem.xai" ->"ecosystem" should not be part of the id itself IMO - if we *can* still change this behavior we should

Reply:
Makes sense, I'll ask winston about it